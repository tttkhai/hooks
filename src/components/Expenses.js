import React from "react";
import ListItem from './ListItem';

function Expenses({expenses}) {
  return (
    <div>
      <div className="expenses">
        {expenses.length === 0 ? (
          <p>Nothing</p>
        ) : (
          expenses.map((res) => {
            return <ListItem key={res.id} className="hello" {...res} />;
          })
        )}
      </div>
    </div>
  );
}

export default Expenses;
