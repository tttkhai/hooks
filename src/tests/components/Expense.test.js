import React from "react";
import { shallow } from "enzyme";
import Expenses from "../../components/Expenses";
import expenses from "../fixtures/text";

test("should render Expense component dynamically", () => {
  const wrapper = shallow(<Expenses expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("test result at class expenses", () => {
  const wrapper = shallow(<Expenses expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render Expense with empty message", () => {
  const wrapper = shallow(<Expenses expenses={[]} />);
  expect(wrapper.find(".expenses p").text()).toBe("Nothing");
});
