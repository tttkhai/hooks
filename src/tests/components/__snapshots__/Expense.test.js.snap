// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense component dynamically 1`] = `
<div>
  <div
    className="expenses"
  >
    <ListItem
      className="hello"
      description="Hello"
      id={1}
      key="1"
    />
    <ListItem
      className="hello"
      description="Testing"
      id={2}
      key="2"
    />
  </div>
</div>
`;

exports[`test result at class expenses 1`] = `
<div>
  <div
    className="expenses"
  >
    <p>
      Nothing
    </p>
  </div>
</div>
`;
